src/main/kotlin/com/team4099/robot2023/subsystems/vision/Vision.kt

```diff
@@ -1,18 +1,286 @@
 package com.team4099.robot2023.subsystems.vision
 
-import com.team4099.lib.vision.VisionMeasurement
+import com.team4099.lib.hal.Clock
+import com.team4099.lib.vision.PoseEstimator
 import com.team4099.robot2023.config.constants.FieldConstants
 import com.team4099.robot2023.config.constants.VisionConstants
+import com.team4099.robot2023.subsystems.vision.camera.CameraIO
+import com.team4099.robot2023.util.PolynomialRegression
+import edu.wpi.first.math.VecBuilder
 import edu.wpi.first.wpilibj2.command.SubsystemBase
 import org.littletonrobotics.junction.Logger
-import org.team4099.lib.apriltag.AprilTagFieldLayout
+import org.team4099.lib.geometry.Pose2d
 import org.team4099.lib.geometry.Pose3d
 import org.team4099.lib.geometry.Pose3dWPILIB
+import org.team4099.lib.geometry.Quaternion
+import org.team4099.lib.geometry.Rotation3d
+import org.team4099.lib.units.base.Time
+import org.team4099.lib.units.base.inMeters
+import org.team4099.lib.units.base.inMilliseconds
+import org.team4099.lib.units.base.inSeconds
+import org.team4099.lib.units.base.meters
 import org.team4099.lib.units.base.seconds
+import org.team4099.lib.units.derived.radians
+import java.util.function.Consumer
+import java.util.function.Supplier
 
-class Vision(cameras: VisionIO) : SubsystemBase() {
-  val ambiguityThreshold = 0.15
-  val targetLogTime = 0.05.seconds
-  val cameraPoses = VisionConstants.CAMERA_TRANSFORMS
+class Vision(vararg cameras: CameraIO) : SubsystemBase() {
+  val io: List<CameraIO> = cameras.toList()
+  val inputs = List(io.size) { CameraIO.CameraInputs() }
 
+  companion object {
+    val ambiguityThreshold = 0.15
+    val targetLogTime = 0.05.seconds
+    val cameraPoses = VisionConstants.CAMERA_TRANSFORMS
+
+    val xyStdDevModel =
+      PolynomialRegression(
+        doubleArrayOf(
+          0.752358,
+          1.016358,
+          1.296358,
+          1.574358,
+          1.913358,
+          2.184358,
+          2.493358,
+          2.758358,
+          3.223358,
+          4.093358,
+          4.726358
+        ),
+        doubleArrayOf(
+          0.005, 0.0135, 0.016, 0.038, 0.0515, 0.0925, 0.0695, 0.046, 0.1245, 0.0815, 0.193
+        ),
+        1
+      )
+
+    val thetaStdDevModel =
+      PolynomialRegression(
+        doubleArrayOf(
+          0.752358,
+          1.016358,
+          1.296358,
+          1.574358,
+          1.913358,
+          2.184358,
+          2.493358,
+          2.758358,
+          3.223358,
+          4.093358,
+          4.726358
+        ),
+        doubleArrayOf(
+          0.008, 0.027, 0.015, 0.044, 0.04, 0.078, 0.049, 0.027, 0.059, 0.029, 0.068
+        ),
+        1
+      )
+  }
+
+  private var poseSupplier = Supplier<Pose2d> { Pose2d() }
+  private var visionConsumer: Consumer<List<PoseEstimator.TimestampedVisionUpdate>> = Consumer {}
```

Comment:
This can be written more cleanly as `private var visionConsumer: (List<PoseEstimator.TimestampedVisionUpdate>) -> Unit = {}`

Reply:
I am not a fan of the `Consumer` and `Supplier` notation bc it is super java specific -- every other language just writes it like this